mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMavericksViewModel.kt

```diff
@@ -146,46 +160,335 @@ abstract class BaseMavericksViewModel<S : MvRxState>(
+                .flowWhenStarted(lifecycleOwner)
+                .distinctUntilChanged()
+                .onEach { lastDeliveredStates[deliveryMode.subscriptionId] = it }
+        } else {
+            flowWhenStarted(lifecycleOwner)
+        }
+        return viewModelScope.launch {
+            flow.collectLatest(action)
+        }.cancelOnClear(lifecycleOwner?.lifecycleScope)
+    }
 
-/**
- * The subscription will receive the most recent state update when transitioning from locked to unlocked states (stopped -> started),
- * only if the state has changed while locked.
- *
- * Likewise, when a MvRxView resubscribes after a configuration change the most recent update will only be emitted
- * if the state has changed while locked.
- *
- * @param subscriptionId A uniqueIdentifier for this subscription. It is an error for two unique only subscriptions to
- * have the same id.
- */
-class UniqueOnly(val subscriptionId: String) : DeliveryMode()
+    @Suppress("EXPERIMENTAL_API_USAGE")
+    private fun <T> Flow<T>.assertOneActiveSubscription(owner: LifecycleOwner, deliveryMode: UniqueOnly): Flow<T> {
+        val observer = object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                if (activeSubscriptions.contains(deliveryMode.subscriptionId)) error(duplicateSubscriptionMessage(deliveryMode))
+                activeSubscriptions += deliveryMode.subscriptionId
+            }
+
+            override fun onDestroy(owner: LifecycleOwner) {
+                activeSubscriptions.remove(deliveryMode.subscriptionId)
+            }
+        }
+
+        owner.lifecycle.addObserver(observer)
+        return onCompletion {
+            activeSubscriptions.remove(deliveryMode.subscriptionId)
+            owner.lifecycle.removeObserver(observer)
+        }
+    }
+
+    private fun <T> lastDeliveredValue(deliveryMode: UniqueOnly): T? {
+        @Suppress("UNCHECKED_CAST")
+        return lastDeliveredStates[deliveryMode.subscriptionId] as T?
+    }
+
+    private fun duplicateSubscriptionMessage(deliveryMode: UniqueOnly) = """
+        Subscribing with a duplicate subscription id: ${deliveryMode.subscriptionId}.
+        If you have multiple uniqueOnly subscriptions in a MvRx view that listen to the same properties
+        you must use a custom subscription id. If you are using a custom MvRxView, make sure you are using the proper
+        lifecycle owner. See BaseMvRxFragment for an example.
+    """.trimIndent()
+
+    private operator fun CoroutineContext.plus(other: CoroutineContext?) = if (other == null) this else this + other
+
+    private fun <S : MvRxState> assertSubscribeToDifferentViewModel(viewModel: BaseMavericksViewModel<S>) {
+        require(this != viewModel) {
+            "This method is for subscribing to other view models. Please pass a different instance as the argument."
+        }
+    }
+
+    private fun Job.cancelOnClear(scope: CoroutineScope?): Job {
+        scope ?: return this
```

Comment:
since you always return "this", you can remove this line and use a safe call on scope below?